### DISCUSSION TITLE
How can i connect powersyc with other backend with api

### DISCUSSION THREAD

**Discussion by shiwam77** [powersync-ja/powersync.dart#50] [Q&A]
I don't want to directly write in HostedDatabase ,I will communicate with api then update the local & in offline fetch from local else from remote. If have these feature then plz give me one demo with flutter so that i will switch to Powersync

**Comment by DominicGBauer** [comment]
Hi @shiwam77 I'm not sure I completely understand what you are asking so want to confirm this is what you mean.

You want to use an API to handle updates to the DB and to use PowerSync to handle synchronising by communicating with the API to update the local on device sqlite DB?

**Comment by shiwam77** [comment]
Mean Every time I have a internet ,
1.For GET request : data will be fetch from remote Server (With API) else from local DB.
2.For POST request : Data will be save in Remote server with API after success then Update local Database,If no internet then put it in batch .

I don't want to synch directly with HostedDatabase rather than i perfer communicating with API,then manage the local Database

**Reply by kobiebotha** [reply]
Hi @shiwam77 

For 1, PowerSync is a sync layer for Postgres - it can not sync from arbitrary APIs and we don't currently have plans to support this.
For 2, this is more or less how PowerSync currently works for writing data back to your systems (PowerSync updates the local db first, then uploads changes)
